src/utils/styles.js

```diff
@@ -6,102 +6,98 @@ const reTranslate = /((^|\s)translate(3d|X)?\()(\-?[\d]+)/;
 
 const reSkew = /((^|\s)skew(x|y)?\()\s*(\-?[\d]+)(deg|rad|grad)(,\s*(\-?[\d]+)(deg|rad|grad))?/;
 
-export default {
 
-  merge() {
-    return ImmutabilityHelper.merge.apply(this, arguments);
-  },
+// This function ensures that `style` supports both ltr and rtl directions by checking
+//   `styleConstants` in `muiTheme` and replacing attribute keys if necessary.
+function ensureDirection(muiTheme, style) {
+  if (process.env.NODE_ENV !== 'production') {
+    warning(!style.didFlip, `You're calling ensureDirection() on the same style
+      object twice.`);
+
+    style = ImmutabilityHelper.merge({
+      didFlip: 'true',
+    }, style);
+  }
+
+  // Left to right is the default. No need to flip anything.
+  if (!muiTheme || !muiTheme.isRtl) return style;
+
+  const flippedAttributes = {
+    // Keys and their replacements.
+    right: 'left',
+    left: 'right',
+    marginRight: 'marginLeft',
+    marginLeft: 'marginRight',
+    paddingRight: 'paddingLeft',
+    paddingLeft: 'paddingRight',
+    borderRight: 'borderLeft',
+    borderLeft: 'borderRight',
+  };
+
+  let newStyle = {};
+
+  Object.keys(style).forEach(function(attribute) {
+    let value = style[attribute];
+    let key = attribute;
+
+    if (flippedAttributes.hasOwnProperty(attribute)) {
+      key = flippedAttributes[attribute];
+    }
 
-  mergeAndPrefix() {
-    let mergedStyles = ImmutabilityHelper.merge.apply(this, arguments);
-    return AutoPrefix.all(mergedStyles);
-  },
+    switch (attribute) {
```

Comment:
I agree, I haven't modified the `ensureDirection` implementation at all yet. I was planning on reviewing it in more detail in my next PR when I refactor `utils/styles.js`.  

This is actually just moving existing `ensureDirections` out of the export. The diffing view looks very bizarre though.